backend/inference.py

```diff
@@ -0,0 +1,105 @@
+# Copyright 2024 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import os
+import signal
+
+from amlip_py.node.AsyncInferenceNode import AsyncInferenceNode, InferenceReplier
+from amlip_py.types.InferenceSolutionDataType import InferenceSolutionDataType
+
+from aml_model_processing import process_model_data
+
+# Domain ID
+DOMAIN_ID = 166
+
+# Custom inference replier
+class CustomInferenceReplier(InferenceReplier):
+
+    def process_inference(
+            self,
+            inference,
+            task_id,
+            client_id):
+
+        # Parse the JSON string
+        data = json.loads(inference.to_string())
+
+        data = data["data"]
+        global aml_model_predict
+        pred = aml_model_predict(data)
+        inference_solution = InferenceSolutionDataType(json.dumps(pred))
+
+        print(f'Data received from client: {client_id}\n'
+            f' with id: {task_id}\n'
+            f' job: {inference.to_string()}\n'
+            f' inference: {inference_solution.to_string()}')
+        return inference_solution
+
+def main():
+    """Execute main routine."""
+
+    ## Prepare model
+    # Get the path to the Downloads directory
+    downloads_path = os.path.join(os.path.expanduser("~"), "Downloads")
+
+    # Create the full path to the file
+    model_path = os.path.join(downloads_path, "model_.json")
+    training_path = os.path.join(downloads_path, "training_set_.json")
+
+    try:
+        with open(model_path, 'r') as file:
+            json = file.read()
+    except FileNotFoundError:
+        print("File not found.")
```

Comment:
Please specify which file is not found
```suggestion
        print("Model file not found.")
```